secret/secrets_manager_vault.go

```diff
@@ -20,20 +22,36 @@ func NewBasicSecretsManager(c SecretsManagerClient) *BasicSecretsManager {
 
 // CreateSecret creates a new secret.
 func (m *BasicSecretsManager) CreateSecret(ctx context.Context, s NamedSecret) (id string, err error) {
-	return "", errors.New("TODO: implement")
+	out, err := m.client.CreateSecret(ctx, &secretsmanager.CreateSecretInput{
+		Name:         s.Name,
+		SecretString: s.Value,
+	})
+	return *out.ARN, err
 }
 
 // GetValue returns an existing secret's decrypted value.
 func (m *BasicSecretsManager) GetValue(ctx context.Context, id string) (val string, err error) {
-	return "", errors.New("TODO: implement")
+	out, err := m.client.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{SecretId: &id})
+	if err != nil {
+		return "", err
+	}
+	return *out.SecretString, err
 }
 
 // UpdateValue updates an existing secret's value.
-func (m *BasicSecretsManager) UpdateValue(ctx context.Context, id string) error {
-	return errors.New("TODO: implement")
+func (m *BasicSecretsManager) UpdateValue(ctx context.Context, id, val string) error {
```

Comment:
Same thing as above - check for empty ID (an empty value may actually be valid).